display-servers/xlib-display-server/src/xwrap/getters.rs

```diff
@@ -462,7 +474,13 @@ impl XWrap {
                 x if x == &self.atoms.NetWMStateBelow => WindowState::Below,
                 _ => WindowState::Modal,
             })
-            .collect()
+            .collect();
+
+        if maximized {
+            window_states.push(WindowState::Maximized);
+        }
```

Comment:
Can't we condense this in an `if-let` in line 454?
Maybe this would help to make the if in the match arms nicer as well.

I mean, that's not a real blocker, more like aesthetic.

Reply:
I can't think of a way to do it in `if-let` syntax, could you provide an example?

Reply:
My bad, I overlooked, that `window_state` is just declared in line 452...